### ISSUE TITLE
used-before-assignment misses a case

### ISSUE THREAD

**Issue by PCManticore** [PyCQA/pylint#1081]
When we have a global variable, with a given name X and inside a function, we are first checking a condition, followed by a loop assignment, we are missing a used-before-assignment case. A reproducing case would be:

```
x = 24

def test(a):
   if x == 2: # this should trigger used-before-assignment
      for x in [1, 2]:
          pass

test(2)
```
